3D-PLOT: Added the basics for 3d interactive plotting with plotly

lnksz opened the pull request:
>_Before merging, please have a look at #20 to avoid conflict in README.md_

Up to now only the basics are implemented
(Done: Commodity layers with capacity+length info, hub processes; Todo: process visualisation)
Still, it has a huge potential and helped to find the bug solved in [PR13](https://github.com/tum-ens/rivus/pull/13)

I gave `matplotlib` a go with 3D visualization, but plotly simply runs rings round it in my opinion.
![plot_image3](https://user-images.githubusercontent.com/15908859/27768610-db95c514-5f17-11e7-9416-c03989b0514d.png)

lnksz committed:
Added the basics for 3d interactive plotting with plotly

ojdo commented:
At first glance, this looks extremely cool. I will give it a spin next weekend, then give you more thorough feedback.

lnksz committed:
Format docstring

Commit:
Adjusted linelength to pep8, cleaned-up unused code

Commit:
Renamed variables to keep to more consistent variable naming

Commit:
one more line length adjustment

lnksz committed:
Correct typos, small nitpicking

lnksz committed:
Change format of edges, where capacity=0.

lnksz committed:
Change default order of commodity layers and their aspect ratio to mirror real aspect ratio better

lnksz committed:
Fix fig3d hub handling

Avoid trying to access edge in Hubs which are not necessarily present. (No hub process is built out in optimum.)

lnksz committed:
Fix hub lookup

`(0,1) in Hubs.index.values` will not work as .values is a numpy array.

lnksz committed:
Add process visualizaiton, Format code, enhance layout

lnksz committed:
Fix MultiIndexed commodity filtering

Commit:
Extend doc-strings for autodoc
